onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorQAttention.cpp

```diff
@@ -319,12 +360,83 @@ class DmlOperatorQAttention : public DmlOperator
         }
         const DML_OPERATOR_DESC maskSlicedDesc = { DML_OPERATOR_SLICE1, &maskSlicedOperatorDesc};
 
+        // We need to slice Past to get PastValue and PastKey tensors for MHA
```

Comment:
:( this is really annoying that we have to do this. It means that we're joining the past key/value in the model just to eventually split them over here. I forgot how different the definitions for `Attention` and `MultiHeadAttention` are. It can really have an impact for larger contexts when you have a cache of sequence length > 2048.

No action to take here, but it may be worth adding an Attention op to DML if it becomes a problem (or adding a QMultiHeadAttention op to ORT).